### DISCUSSION TITLE
Using Sdbus-c++ has dependency to Server Start way?

### DISCUSSION THREAD

**Discussion by Allen1218** [Kistler-Group/sdbus-cpp#268] [General]
## Environment description
- sdbus C++ version  ： 0.8.3
- Using sdbus-c++ system, not session.

##  Quenstion decription
Start Server way --- "./server ", then client and server coommunicate is good.

But another start server way --- "./server &", then client and server communicate is bad. Meanwhile, in this case, I using "dbus-monitor --system" that can not catch the bus message.

## sdbus server code program 
sdbus server has coded by multi-thread, then connection is start in main thread, server method receicve client message in another thread.

## Looking forward to you reply!
In my case, must using muti-thread in sdbus client communicate with server.  would you tell me, how to this question  reason?
Thanks very much!

**Comment by Allen1218** [comment]
using UDL-xml generate code.

**Comment by kodiakz** [comment]
Does this issue still persist? How do you start your server and client? Two different terminals? In case of running in background with **&** within the same terminal?
From your description I can not see how running the server in background or in foreground could change the runtime behavior of sdbus-c++.
